devops/openshift/db-migrations-job.yml

```diff
@@ -0,0 +1,87 @@
+apiVersion: template.openshift.io/v1
+kind: Template
+metadata:
+  name: ${NAME}
+  annotations:
+    description: "Template for running a job that performs a database migrations"
+labels:
+  project: ${PROJECT}
+  service: ${SERVICE_NAME}
+parameters:
+  - name: NAME
+    required: true
+  - name: BUILD_TAG
+    value: "0"
+  - name: PROJECT
+    value: sims
+  - name: SERVICE_NAME
+    value: db-migrations
+  - name: JOB_NAME
+    description: database.migrations
+    required: true
+    value: db-migrations-job
+  - name: IMAGE_STREAM_TAG
+    required: true
+  - name: BUILD_NAMESPACE
+    required: true
+  - name: IMAGE_REGISTRY
+    value: image-registry.openshift-image-registry.svc:5000
+  - name: DB_SERVICE
+    value: patroni-master
+  - name: DB_SECRET_NAME
+    value: patroni-creds
+  - name: DB_USERNAME_KEY
+    value: superuser-username
+  - name: DB_PASSWORD_KEY
+    value: superuser-password
+  - name: DATABASE_NAME_KEY
+    required: true
+  - name: CPU_LIMIT
+    value: "1.0"
+  - name: MEMORY_LIMIT
+    value: "2000M"
+  - name: CPU_REQUEST
+    value: "0.5"
+  - name: MEMORY_REQUEST
+    value: "1000M"
+objects:
+  - apiVersion: batch/v1
+    kind: Job
+    metadata:
+      name: ${JOB_NAME}
+    spec:
+      template:
+        metadata:
+          name: ${JOB_NAME}
+        spec:
+          containers:
+            - name: db-migrations-container
```

Comment:
Timeout and retries? 

![image](https://github.com/bcgov/SIMS/assets/54600590/d72c3f5c-f308-42b9-ba66-cf663dd042cc)

Reply:
Yes the job has the maximum duration the job can run and number of retries, I've started handling it in the "wait-for-db-migrations" of makefile by checking the status of the job every 5 sec and so did not add them here, it will be a duplicate and during the github actions run, the failure is shown in the logs.